test/unit/api_spec.js

```diff
@@ -1903,6 +1903,46 @@ describe("api", function () {
       await loadingTask.destroy();
     });
 
+    it("write a value in an annotation, save the pdf and check the value in xfa datasets", async function () {
+      if (isNodeJS) {
+        pending("Linked test-cases are not supported in Node.js.");
+      }
+
+      let loadingTask = getDocument(buildGetDocumentParams("issue16081.pdf"));
+      let pdfDoc = await loadingTask.promise;
+      const value = "Hello World";
+
+      pdfDoc.annotationStorage.setValue("2055R", { value });
+
+      const data = await pdfDoc.saveDocument();
+      await loadingTask.destroy();
+
+      loadingTask = getDocument(data);
+      pdfDoc = await loadingTask.promise;
+      const datasets = await pdfDoc.getXFADatasets();
+
+      const getNamedNode = (node, path) => {
+        for (const component of path.split(".")) {
+          if (!node.childNodes) {
+            break;
+          }
+          for (const child of node.childNodes) {
+            if (child.nodeName === component) {
+              node = child;
+              break;
+            }
+          }
+        }
+        return node;
+      };
+
+      const surName = getNamedNode(
+        datasets.node,
+        "xfa:data.PPTC_153.Page1.PersonalInformation.TitleAndNameInformation.PersonalInfo.Surname.#text"
+      );
+      expect(surName.nodeValue).toEqual(value);
```

Comment:
Please remember to clean-up everything at the end:
```suggestion
      expect(surName.nodeValue).toEqual(value);
      
      await loadingTask.destroy();
```